deployment/ccip/changeset/cs_chain_contracts.go

```diff
@@ -1078,13 +1076,22 @@ func UpdateOffRampSourcesChangeset(e deployment.Environment, cfg UpdateOffRampSo
 			} else {
 				router = state.Chains[chainSel].Router.Address()
 			}
-			onRamp := state.Chains[source].OnRamp
+			sourceChainFamily, _ := chain_selectors.GetSelectorFamily(source)
+
+			onRampBytes := []byte{}
+			if sourceChainFamily == chain_selectors.FamilyEVM {
+				onRampBytes = state.Chains[source].OnRamp.Address().Bytes()
+			} else if sourceChainFamily == chain_selectors.FamilySolana {
+				// for solana the router contract has the onramp logic
+				onRampBytes = state.SolChains[source].Router.Bytes()
+			}
+
```

Comment:
let's have a method on state to return this something like - https://github.com/smartcontractkit/chainlink/blob/develop/deployment/ccip/changeset/state.go#L166C25-L166C41